### DISCUSSION TITLE
Does sdrpp have an API?

### DISCUSSION THREAD

**Discussion by zeroby0** [AlexandreRouma/SDRPlusPlus#741] [Q&A]
Hi!

I want to embed sdrpp, and automate recording, decoding, and drawing NOAA and Meteor transmissions.

Since there is no command line interface to control sdrpp, I'm wondering if there's any other ways to do this. I found the server mode, but couldn't find any docs for it's API, or information about how it works.

Btw, thank you SO much for creating sdrpp! It's absolutely amazing! Especially the DSP and the inbuilt modules!

**Comment by zeroby0** [comment]
Maybe I could build with rigctl loaded and enabled by default, and use rigctl to control it. Although I'm yet to find out the extent to which rigctl can control sdrpp.

**Reply by gvanem** [reply]
Take a look at [PyWinAuto](https://github.com/pywinauto/pywinauto).

**Reply by zeroby0** [reply]
Looks very cool!

Personally, I'd like to run sdrpp without a GUI on a Raspberry Pi, but this would be very useful for windows peeps.

**Reply by AlexandreRouma** [reply]
you can't run SDR++ without the GUI. The server mode doesn't do demodulation.
I suggest you either used the DSP library on its own or used some other premade NOAA/Meteor receiver like AltiWX
